docs/reference/rollup/apis/rollup-api.asciidoc

```diff
@@ -147,36 +151,44 @@ object, this property is required.
 `terms`::
 (Optional, <<search-aggregations-bucket-terms-aggregation,`terms`
 aggregation>> object)
-Groups and stores unique values for <<keyword,keyword family>> and
-<<number,numeric>> fields.
+Groups documents based on unique field values.
 +
 .Properties of `terms`
 [%collapsible%open]
 ======
 `fields`::
-(Required*, string or array of strings)
-<<keyword,Keyword family>> and <<number,numeric>> fields to store. If you
-specify a `terms` object, this property is required.
 +
+--
+(Required*, string or array of strings)
+Fields to store unique values for. Supports the following field types:
+
+* <<keyword,Keyword family>> types
+* <<number,Numeric>> types, excluding `unsigned_long`
+* <<text,`text`>> fields with <<fielddata-mapping-param,`fielddata`>> enabled
+
+If you specify a `terms` object, this property is required.
+
 TIP: Avoid storing high-cardinality fields. High-cardinality fields can greatly
 increase the size of the resulting rollup index.
+--
 ======
 =====
 
 `metrics`::
 (Required, object or array of objects)
-Collects and stores metrics for <<number,numeric>> fields. You must specify at
-least one `metrics` object.
+Collects and stores metrics for fields. You must specify at least one `metrics`
+object.
 +
 .Properties of `metrics` objects
 [%collapsible%open]
 =====
 `field`::
 (Required, string)
-<<number,Numeric>> field to collect metrics for.
+<<number,Numeric>>, <<date,`date`>>, or <<date_nanos,`date_nanos`>> field to
+collect metrics for. `unsigned_long` fields are not supported.
```

Comment:
There seems to be a minor bug with `date_nanos` fields. Here's reproduction steps:

```
// Map a date_nanos field.
PUT my-index-000001
{
  "mappings": {
    "properties": {
      "my-date-nanos-field": {
        "type": "date_nanos" 
      }
    }
  }
}

// Index a document with the date_nanos field.
POST my-index-000001/_doc
{ 
  "@timestamp": "2099-11-15T13:12:00",
  "my-date-nanos-field": "2015-01-01T12:10:30.123456789Z"
}

// Attempt to roll up the index with the date_nanos field as a metric.
POST my-index-000001/_rollup/rollup-my-index-000001
{
  "groups": {
    "date_histogram": {
      "field": "my-date-nanos-field",
      "calendar_interval": "1d"
    }
  },
  "metrics": [
    {
      "field": "@timestamp",
      "metrics": [
        "min",
        "max"
      ]
    }
  ]
}
```

The rollup attempt returns an error:

```
{
  "error" : {
    "root_cause" : [
      {
        "type" : "illegal_argument_exception",
        "reason" : "trying to convert a known date time formatter to a nanosecond one, wrong field used?"
      }
    ],
    "type" : "exception",
    "reason" : "Unable to rollup index [my-index-000001]",
    "caused_by" : {
      "type" : "exception",
      "reason" : "[my-index-000001/9DveCFw8RR67S2tul3Qt_w][[my-index-000001][0]] org.elasticsearch.action.support.broadcast.BroadcastShardOperationFailedException:",
      "caused_by" : {
        "type" : "illegal_argument_exception",
        "reason" : "trying to convert a known date time formatter to a nanosecond one, wrong field used?"
      }
    }
  },
  "status" : 500
}
```

Reply:
^^ @talevy

Reply:
I will take a look. thank you!

Reply:
I opened https://github.com/elastic/elasticsearch/issues/70161